db/compaction/compaction_picker.cc

```diff
@@ -39,32 +39,46 @@ bool FindIntraL0Compaction(const std::vector<FileMetaData*>& level_files,
                            size_t min_files_to_compact,
                            uint64_t max_compact_bytes_per_del_file,
                            uint64_t max_compaction_bytes,
-                           CompactionInputFiles* comp_inputs) {
-  size_t compact_bytes = static_cast<size_t>(level_files[0]->fd.file_size);
-  uint64_t compensated_compact_bytes = level_files[0]->compensated_file_size;
+                           CompactionInputFiles* comp_inputs,
+                           SequenceNumber earliest_mem_seqno) {
+
+  // Do not pick ingested file when there is at least one memtable not flushed which of seqno is overlap with the sst.
+  size_t start = 0;
+  while (start < level_files.size() && level_files[start]->being_compacted &&
+      level_files[start]->fd.smallest_seqno == level_files[start]->fd.largest_seqno &&
```

Comment:
@Little-Wallace I realize that the check for `smallest_seqno == largest_seqno` is identifying ingested sstables, but I'm not sure we actually need or want that. If there is some other way for an L0 sstable to get created where `sstable.largest_seqno >= earliest_mem_seqno` we wouldn't want to involve that sstable in an intra-L0 compaction.